src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CrlCache.cs

```diff
@@ -91,13 +91,21 @@ private static bool AddCachedCrl(string crlFileName, SafeX509StoreHandle store,
                         return false;
                     }
 
+                    IntPtr nextUpdatePtr = Interop.Crypto.GetX509CrlNextUpdate(crl);
+
+                    // If there is no crl.NextUpdate, this indicates that the CA is not providing
+                    // any more updates to the CRL.
+                    if (nextUpdatePtr == IntPtr.Zero)
+                    {
+                        return false;
```

Comment:
Rather than permanency, can we do something like

```C#
DateTime nextUpdate;

if (nextUpdatePtr == IntPtr.Zero)
{
    nextUpdate = GetLastModifiedTime(crlFile).AddDays(3);
}
else
{
    nextUpdate = OpenSslX509CertificateReader.ExtractValidityDateTime(nextUpdatePtr);
}
```

?

It would make me feel more comfortable that we don't forever-cache a clerical error, while doing some amount of caching assuming it's intentional/permanent.

Reply:
That is a good idea.